routes/index.js

```diff
@@ -260,4 +275,70 @@ router.post('/register', (req, res, next) =>{
     }
 })
 
+router.get('/preferences', common.isAuthenticated, (req, res, next) => {
+    var data = { title : 'Preferences'}
+    res.render('preferences', data);
+})
+
+router.get('/gettingstarted', common.isAuthenticated, (req, res, next) => {
+    var data = { title: 'Getting Started'}
+    res.render('gettingstarted', data);
+})
+
+router.get('/faq', common.isAuthenticated, (req, res, next) => {
+    var data = { title: 'Frequently Asked Questions'}
+    res.render('faq', data);
+})
+
+router.get('/updateaccount', common.isAuthenticated, (req, res, next) => {
+    var payload = { title : 'Update Account Settings' }
+    var data = [];
+    //Select all information from database. Render form with values all as the existing information in users table.
+    sql.pool.query('SELECT * FROM Users WHERE id=?', [req.session.userId],
+    (err, results) => {
+        if (err) {
+            if (err) {
+                req.flash('error', err)
+                res.render('/auth/updateaccount', data)
+            }
+        } else {
+            data.push(new Account(results[0].firstName, results[0].lastName, results[0].email, results[0].address, results[0].city, results[0].state, results[0].postalCode, results[0].country, results[0].worldwide, results[0].aboutMe));
+        }
+        payload.data = data;
+        res.render('auth/updateaccount', payload);
+    });
+})
+
+router.post('/updateaccount', common.isAuthenticated, (req, res, next) => {
+    var data = {
+        title: 'Update Account Settings',
+        userId: req.session.userId,
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        email: req.body.email,
+        address: req.body.address,
+        city: req.body.city,
+        state: req.body.state,
+        postalCode: req.body.postalCode,
+        country: req.body.country,
+        worldwide: req.body.worldwide,
+        aboutMe: req.body.aboutMe
+    }
+    sql.pool.query('UPDATE Users SET firstName=?, lastName=?, email=?, address=?, city=?, state=?, postalCode=?, country=?, worldwide=?, aboutMe=? WHERE id=?',
```

Comment:
Could we change this to an UPDATE Users SET ? where id = ? instead so it's not such a lengthy query. You can add the title to the data just before rendering so it doesn't get sent with the sql data as that will crash it.

Reply:
We should also probably first do a check to see if the email already exist for a user where id <> current user id. If it already exists then we can send back a more meaningful error message before trying to do the full update. You'd still use the req.flash('error', 'Email already exists') or something like that and render back the page with the data so the fields are filled in and they don't have to re-enter all changes they made.

Reply:
The password is not brought into the account settings form, so just changing it to what you had didn't work. I have moved the title out of data. I didn't try to mess around with it that much but do I need to grab the password from the database first, add it to data, and then send the update request?

Reply:
So the SET will only update the properties you have in the json object, it doesn't have to have everything. The reason that simply removing the title out didn't work is because you still have the userId in there which is not a valid field in the users table. You would need to remove that from the data object and just pass it in as a separate parameter for the where. I'll approve this with the way it is but I think we should try to take advantage of this functionality where possible rather than passing in a bunch of separate parameters. The query would then be UPDATE Users SET ? WHERE id = ? [data, req.session.userId] and the data would no longer have the user id or title in it.

Did you decide to skip the check to see if the email already exists? I think this should be done and we should send back a more meaningful display message with flash if the email already exists rather than sending everything to sql and letting it error with something that won't mean much.

Reply:
That makes sense. I'll make the change this week. I could refactor this as part of the assignment's code smell